### PR TITLE
Sara/mkt 440 update descriptions for docs displayed

### PR DESCRIPTION
**Pull request by s-santillan**
# Thanks for improving Semgrep Docs 😀

### Please ensure

(n/a) A subject matter expert (SME) reviews the content

- [ ] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team
- [ ] Redirects are added if the PR changes page URLs

Requesting the whole cavalry for this: @pabloest and @adamkvitek 
### Approach

Refer to the [linear task](https://linear.app/r2c/issue/MKT-440/update-descriptions-for-docs-displayed-in-card-index) to understand the goal and approach behind this task.

### A note on the CLI docs:

I went through the CLI docs, which have not been update for a while. You can see in the changes that I added the following depending on what was missing:

* `slug`
* `hide_title: true` (this is so that the title in the sidebar can be different from the title in the document, which we do sometimes to expand acronyms in the document but not in the sidebar)
* `append_help_link`
* `description`
* `title`

**Just skimming some of them reveals potential regressions**. Not sure if we should prioritize it, it'll take time to piece out how to modularize, etc.

### CLI:
CLI

#### CI:
CI

### CODE REVIEW

File: docs/running-rules.md [hunk 1]
```diff
@@ -1,6 +1,9 @@
 ---
 slug: running-rules
 append_help_link: true
+description: "Learn about Semgrep rules,how to add your custom rules and rules from Semgrep Registry, a community-contributed repository of rules to help enforce security."
```
Comment (adamkvitek) [inline docs/running-rules.md:4]:
```suggestion
description: "Learn about Semgrep rules, how to add your custom rules and rules from Semgrep Registry, a community-contributed repository of rules to help enforce security."
```

**Review by adamkvitek — APPROVED** [review]
@s-santillan thank you! Looks great, left one tiny suggestion. The whole diff shows very good changes!

File: docs/troubleshooting/gitlab-sast.md [hunk 2]
```diff
@@ -3,9 +3,8 @@ slug: gitlab-sast
 title: Troubleshooting GitLab SAST
 hide_title: true
 description: >-
-  GitLab SAST includes an analyzer that runs Semgrep.
-  Fix issues with semgrep-sast jobs running slowly,
-  not showing results, or erroring.
+  Fix issues with GitLab SAST's Semgrep analyzer such as jobs running slowly,
```
Comment (pabloest) [inline docs/troubleshooting/gitlab-sast.md:6]:
```suggestion
  Fix issues with GitLab SAST's Semgrep analyzer, such as jobs running slowly,
```

File: docs/troubleshooting/semgrep.md [hunk 3]
```diff
@@ -1,6 +1,9 @@
 ---
 slug: semgrep
-description: "Ways you can get more information when semgrep CLI or CI hangs, crashes, or just takes too long."
+description: "Get more information when Semgrep CLI or CI hangs, crashes, times out, or is slow."
```
Comment (pabloest) [inline docs/troubleshooting/semgrep.md:3]:
```suggestion
description: "Get more information when Semgrep hangs, crashes, times out, or is slow."
```

**Review by pabloest — APPROVED** [review]

**Comment by s-santillan** [comment]
thanks @pabloest and @adamkvitek!

**Pull request merged**